cmd/commandfuncs.go

```diff
@@ -588,16 +591,9 @@ func cmdUpgrade(_ Flags) (int, error) {
 	if err != nil {
 		return caddy.ExitCodeFailedStartup, fmt.Errorf("unable to enumerate installed plugins: %v", err)
 	}
-	pluginPkgs := make(map[string]struct{})
-	for _, mod := range nonstandard {
-		if mod.goModule.Replace != nil {
-			return caddy.ExitCodeFailedStartup, fmt.Errorf("cannot auto-upgrade when Go module has been replaced: %s => %s",
-				mod.goModule.Path, mod.goModule.Replace.Path)
-		}
-		l.Info("found non-standard module",
-			zap.String("id", mod.caddyModuleID),
-			zap.String("package", mod.goModule.Path))
-		pluginPkgs[mod.goModule.Path] = struct{}{}
+	pluginPkgs, err := buildPlugins(nonstandard)
```

Comment:
Maybe call this function `getPluginPackages`? `build` is ambiguous here, my first thought is this would attempt to "build" at this point, i.e. compile a build of Caddy. Instead, it just makes a map, with some validation.